From f77bcee53f556fdf41c2d4405f231d612af62ae5 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Marc-Andr=C3=A9=20Lureau?= <marc-andre.lureau@nokia.com>
Date: Tue, 10 Mar 2009 18:01:44 +0200
Subject: [PATCH 24/75] pulsecore: forward flat_volume functions to flat_volume_sink

btw, pa_sink_propagate_flat_volume() could be static... in that case,
we could make sure it's not called with a slaved sink.
---
 src/pulsecore/sink.c |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/src/pulsecore/sink.c b/src/pulsecore/sink.c
index 930a87d..02405cb 100644
--- a/src/pulsecore/sink.c
+++ b/src/pulsecore/sink.c
@@ -1110,6 +1110,13 @@ void pa_sink_update_flat_volume(pa_sink *s, pa_cvolume *new_volume) {
      * the sinks volume here, we only return how it needs to be
      * updated. The caller should then call pa_sink_set_volume().*/
 
+    if (s->flat_volume_sink) {
+        /* this sink is flatten with flat_volume_sink */
+        pa_sink_update_flat_volume(s->flat_volume_sink, new_volume);
+        pa_cvolume_remap(new_volume, &s->flat_volume_sink->channel_map, &s->channel_map);
+        return;
+    }
+
     if (pa_idxset_isempty(s->inputs)) {
         /* In the special case that we have no sink input we leave the
          * volume unmodified. */
@@ -1185,6 +1192,12 @@ void pa_sink_propagate_flat_volume(pa_sink *s) {
     /* This is called whenever the sink volume changes that is not
      * caused by a sink input volume change. We need to fix up the
      * sink input volumes accordingly */
+    if (s->flat_volume_sink) {
+        /* this sink is flatten with flat_volume_sink */
+        pa_sink_propagate_flat_volume(s->flat_volume_sink);
+        return;
+    }
+
     propagate_flat_volume(s, s);
 }
 
-- 
1.6.3.3

